src/lib/thumbnails.js

```diff
@@ -0,0 +1,87 @@
+import fs from 'fs'
+import { ImagePool } from '@squoosh/lib'
+
+const episode_thumbnail_dir = 'public/assets/podcast/thumbnails'
+const episode_thumbnails_url_path = '/assets/podcast/thumbnails'
+
+const supported_thumbnail_formats = ['jpg', 'webp']
+const default_size = 400
+
+const toEpisodeThumbnailFileName = (episode, size, format) =>
+  `s${episode.season}e${episode.episode}_${size}x${size}.${format}`
+
+const toThumbnailFileNamesByFormat = (episode, size, formats) =>
+  Object.fromEntries(formats.map((format) => [format, toEpisodeThumbnailFileName(episode, size, format)]))
+
+const fileExists = async (file) =>
+  fs.promises
+    .access(file)
+    .then(() => true)
+    .catch(() => false)
+
+const fetchRawImage = async (imageUrl) => (await fetch(imageUrl)).buffer()
+
+const createThumbnailsFromImageBufferPool = async (imagePool, imageBuffer, size) => {
+  const image = imagePool.ingestImage(imageBuffer)
+
+  await image.decoded
+  await image.preprocess({
+    resize: {
+      enabled: true,
+      width: size,
+    },
+  })
+
+  await image.encode({
+    webp: {},
+    mozjpeg: {},
+  })
+
+  return {
+    jpg: (await image.encodedWith.mozjpeg).binary,
+    webp: (await image.encodedWith.webp).binary,
+  }
+}
+
+export async function downloadAllMissingEpisodeThumbnails(episodes, size = default_size) {
+  const imagePool = new ImagePool()
+
+  for (const episode of episodes) {
+    const thumbnailFileNamesByFormat = toThumbnailFileNamesByFormat(episode, size, supported_thumbnail_formats)
+
+    const allThumbnailsPresent = await Object.values(thumbnailFileNamesByFormat)
+      .map(async (fileName) => await fileExists(`${episode_thumbnail_dir}/${fileName}`))
+      .reduce((a, b) => a && b, true)
```

Comment:
The `map` call returns an array of promises here so you need to wait for them inside reduce. Or wait for them all at once and then reduce on the resolved results.

I think something like this should work. It's maybe not the solution that's easiest on the eye. Just to illustrate what I mean:

```suggestion
      .reduce(async (a, b) => (await a) && (await b), true)
```

Reply:
You are completely right. Strange, that it does what it is supposed to..

Would
```js
const allThumbnailsPresent = await Promise.all(Object.values(thumbnailFileNamesByFormat)
      .map(async (fileName) => await fileExists(`${episode_thumbnail_dir}/${fileName}`)))
      .reduce((a, b) => a && b, true)
```
be an acceptable adaption?